qiskit/assertions/asserts.py

```diff
@@ -0,0 +1,120 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Abstract class and all derived classes for making statistical assertions.
+"""
+import abc
+from datetime import datetime
+from qiskit.circuit.register import Register
+from qiskit.circuit.classicalregister import ClassicalRegister, Clbit
+from qiskit.circuit.measure import Measure
+from qiskit.exceptions import QiskitError
+
+
+class Asserts(Measure):
+    """A superclass for all assertions, and a subclass of Measure"""
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, qubit, cbit, pcrit, negate, type_str):
+        """
+        Constructor for Asserts
+
+        Args:
+            qubit(QuantumRegister or list or tuple): quantum register
+            cbit(ClassicalRegister or list or tuple): classical register
+            pcrit(float): the critical p-value
+            negate(bool): True if assertion passed is negation of statistical test passed
+            type_str(string): the type of the assertion
+        """
+        super().__init__()
+        self._qubit = qubit
+        self._cbit = cbit
+        self._pcrit = pcrit
+        self._negate = negate
+        self._type = type_str
+        self._expval = None
+
+    @staticmethod
+    def breakpoint_name():
+        """
+        Returns the name of the breakpoint.  A breakpoint is a QuantumCircuit
+        whose last instruction is an assertion instruction.
+
+        Returns:
+            string: the breakpoint name
+        """
+        return "breakpoint_" + datetime.now().isoformat()
+
+    @staticmethod
+    def syntax4measure(bit):
+        """
+        Returns the input qubit or cbit, suitable for the syntax of the measure
+        instruction, as a list.
+
+        Args:
+            bit(QuantumRegister or ClassicalRegister or list or tuple): register
+
+        Returns:
+            list or Register: bit as a list
+        """
+        if isinstance(bit, (list, Register)):
+            return bit
+        elif isinstance(bit, (range, tuple)):
+            return list(bit)
+        else:
+            return [bit]
+
+    @staticmethod
+    def clbits2idxs(cbits, exp):
+        """
+        Returns the indices of cbits with respect to exp, the QuantumCircuit.
+
+        Args:
+            cbits(ClassicalRegister or list or tuple): cbit of a measurement instruction
```

Comment:
bits as tuples is being deprecated.

Reply:
[resolved by this commit](https://github.com/Qiskit/qiskit-terra/pull/2867/commits/e64d7ea155c224da82759685a0c44e59aa11758a)